underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -401,15 +514,25 @@ private boolean isDirectory(String path) throws IOException {
       path = stripPrefixIfPresent(path);
       path = PathUtils.normalizePath(path, PATH_SEPARATOR);
       path = path.equals(PATH_SEPARATOR) ? "" : path;
-      Directory directory = new Directory(path, '/');
+      Directory directory = new Directory(path, PATH_SEPARATOR_CHAR);
       Container container = mAccount.getContainer(mContainerName);
       Collection<DirectoryOrObject> objects = container.listDirectory(directory);
       Set<String> children = new HashSet<>();
+      boolean foundSelf = false;
       for (DirectoryOrObject object : objects) {
         String child = stripFolderSuffixIfPresent(object.getName());
-        String noPrefix = stripPrefixIfPresent(child, path);
-        children.add(noPrefix);
+        String noPrefix = UnderFileSystemUtils.stripPrefixIfPresent(child, path);
+        if (!noPrefix.equals(stripFolderSuffixIfPresent(path))) {
```

Comment:
`stripFolderSuffixIfPresent(path)` should be constant right? If so, this can be evaluated once and stored in a variable.